File: plugins/OrchidForms/src/main/java/com/eden/orchid/presentations/fields/TextField.java
package com.eden.orchid.presentations.fields;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.presentations.FormField;
import org.json.JSONObject;

import javax.inject.Inject;
import java.util.List;

public class TextField extends FormField {

    @Inject
    public TextField(OrchidContext context) {
        super(context);
    }

    public void initialize(String key, JSONObject fieldData) {
        super.initialize(key, fieldData);
    }

    public List<String> getTemplates() {
        List<String> allTemplates = super.getTemplates();
        allTemplates.add("fields/textual.twig");
        return allTemplates;
    }

    public boolean acceptsType(String type) {
        switch(type) {
            case "text":
            case "number":
                return true;
            default:
                return false;
        }
    }

}

File: plugins/OrchidForms/src/main/kotlin/com/eden/orchid/forms/model/fields/TextField.kt
package com.eden.orchid.forms.model.fields

import com.eden.orchid.api.options.annotations.Description
import com.eden.orchid.forms.model.FormField

@Description("A simple text field, which may optionally be any of tht HTML5 textual input types.", name="Text")
class TextField : FormField(arrayOf(
        "text",
        "password",
        "datetime",
        "datetime-local",
        "date",
        "month",
        "time",
        "week",
        "number",
        "email",
        "url",
        "search",
        "tel",
        "color"
)) {

    override fun getTemplates(): List<String> {
        val allTemplates = super.getTemplates().toMutableList()
        allTemplates.add("fields/textual")
        return allTemplates
    }

}
